bin/importGlossary.js

```diff
@@ -97,17 +173,226 @@ const rows = parseElement(documentContext)(table)
+    }
+
+    // Determine if update is needed
+    const needsUpdate =
+      definitionDifferent || aliasesDifferent || imageDifferent;
+
+    // Only show detailed debugging info if DEBUG env var is set
+    if (needsUpdate && process.env.DEBUG) {
+      console.log(`\nDetailed debugging for "${phrase}":`);
+
+      if (definitionDifferent) {
+        console.log(`  Definition differs and will be updated`);
+      }
+
+      if (aliasesDifferent) {
+        console.log(`  Aliases differ and will be updated`);
+      }
+
+      if (imageDifferent) {
+        console.log(`  Image differs and will be updated`);
+      }
+    }
+
+    return {
+      row,
+      existingGlossary,
+      needsUpdate: needsUpdate,
+      existsInCoda: true,
+      existingEntry,
+    };
+  }
+
+  return {
+    row,
+    existingGlossary,
+    needsUpdate: true, // New entry needs to be created
+    existsInCoda: false,
+  };
+});
+
+// Count and report entries needing updates
+const entriesNeedingUpdate = rowsWithMetadata.filter(
+  (r) => r.needsUpdate
+).length;
+const skippableEntries = rowsWithMetadata.filter((r) => !r.needsUpdate).length;
+
+console.log(`\nAnalysis before processing:`);
+console.log(`- ${skippableEntries} entries can be skipped (no changes needed)`);
+console.log(`- ${entriesNeedingUpdate} entries need to be created or updated`);
+
+// Update all glossary entries and track successfully updated terms
+const updateResults = await syncApply(
+  rowsWithMetadata,
+  async (metadata) => await setGlossary(metadata)
 );
+
+// Summarize the results
+const skipped = updateResults.filter((r) => r.skipped).length;
+const updated = updateResults.filter((r) => r.updated).length;
+const failed = updateResults.filter((r) => !r.succeeded).length;
+
+console.log(`\nGlossary update summary:`);
+console.log(`- ${skipped} entries skipped (no changes needed)`);
+console.log(`- ${updated} entries updated or added`);
+console.log(`- ${failed} entries failed to update`);
+console.log(`- ${rows.length} total entries in Google Doc`);
+
+// Only proceed with deletion if ALL entries were successfully updated
+const allSuccessful = updateResults.every((result) => result.succeeded);
+
+if (allSuccessful) {
+  console.log(
+    "\nAll entries were successfully processed. Checking for outdated entries to remove..."
+  );
+
+  // Remove any items that are in Coda but not in the Gdoc
+  // Use a more comprehensive normalization to avoid issues with apostrophes and special chars
+  const terms = rows.map(({ term }) => {
+    const decoded = decode(term.trim());
+    let normalized = decoded.toLowerCase().trim();
+
+    // Handle common special cases that cause comparison issues
+    // Replace apostrophes with empty string for consistent matching
+    normalized = normalized.replace(/['']s\b/g, "s"); // Replace "'s" and "'s" with "s"
+    normalized = normalized.replace(/['']/g, ""); // Remove all apostrophes
```

Comment:
wtf? not sure if the desired business logic was more complex, but what the fluffy code is doing can be written as:
```suggestion
    normalized = normalized.replace(/'/g, "");
```